src/Type/Php/MinMaxFunctionReturnTypeExtension.php

```diff
@@ -64,6 +66,31 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return new ErrorType();
 		}
 
+		// rewrite min($x, $y) as $x < $y ? $x : $y
+		// we don't handle arrays, which have different semantics
+		$functionName = $functionReflection->getName();
+		$args = $functionCall->args;
+		if (count($functionCall->args) === 2) {
+			$argType0 = $scope->getType($args[0]->value);
+			$argType1 = $scope->getType($args[1]->value);
+
+			if ($argType0->isArray()->no() && $argType1->isArray()->no()) {
+				if ($functionName === 'min') {
+					return $scope->getType(new Ternary(
+						new Smaller($args[0]->value, $args[1]->value),
+						$args[0]->value,
+						$args[1]->value
+					));
+				} elseif ($functionName === 'max') {
+					return $scope->getType(new Ternary(
+						new Smaller($args[0]->value, $args[1]->value),
+						$args[1]->value,
+						$args[0]->value
+					));
+				}
+			}
+		}
+
```

Comment:
Isn't it possible that we could delete some code that's already in the extension thanks to this logic?

Reply:
the code above only handles the 1-arg case and the code below handles the any-number-of-args case.

I guess we can drop it, after we push the 

```
return $scope->getType(new Ternary(
						new Smaller($args[0]->value, $args[1]->value),
						$args[1]->value,
						$args[0]->value
					));
```

approach into a direction where it supports any number of args (using `BooleanAnd` and `BooleanOr`) as you suggested in https://github.com/phpstan/phpstan-src/pull/604#discussion_r689521573

but IMO this should be a separate PR

Reply:
Cool :)